src/data/list/basic.lean

```diff
@@ -1584,6 +1584,27 @@ begin
     exact lt_of_add_lt_add_left' (lt_of_le_of_lt h $ add_lt_add_right (lt_of_not_ge h') _) }
 end
 
+-- Several lemmas about sum/head/tail for `list ℕ`.
+-- These are hard to generalize well, as they rely on the fact that `default ℕ = 0`.
+
+-- We'd like to state this as `L.head * L.tail.prod = L.prod`,
+-- but because `L.head` relies on an inhabited instances and
+-- returns a garbage value for the empty list, this is not possible.
+-- Instead we write the statement in terms of `(L.nth 0).get_or_else 1`,
+-- and below, restate the lemma just for `ℕ`.
+@[to_additive]
+lemma head_mul_tail_prod' [monoid α] (L : list α) : (L.nth 0).get_or_else 1 * L.tail.prod = L.prod :=
+by { cases L, { simp, refl, }, { simp, }, }
```

Comment:
@jcommelin, how do you feel about this version?

Reply:
Yeah... that's whay we'll have to live with :wink: 
Thanks for working on this!